### PR TITLE
feat: disable+enable snap service on storage events

### PR DESCRIPTION
**Pull request by marcoppenheimer**
##### `feat: disable+enable snap service on storage events`
- This is needed as new storage mounts aren't found by the snap, even with plug `removable-devices`. This is likely a bug, but am unable to resolve without this workaround

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -59,44 +57,22 @@ async def test_logs_write_to_storage(ops_test: OpsTest):
...
-        servers=servers,
-        username=username,
-        password=password,
-        topic=topic,
-        consumer_group_prefix=None,
-        security_protocol=security_protocol,
+@pytest.mark.abort_on_fail
+async def test_logs_write_to_new_storage(ops_test: OpsTest):
+    check_output(f"JUJU_MODEL={ops_test.model_full_name} juju add-storage kafka/0 log-data")
+    time.sleep(10)  # to give time for storage to complete
```
Comment (welpaolo) [inline tests/integration/test_charm.py:71]: Is there any way in which you can ensure that the storage is attached correctly ?

**Review by welpaolo — COMMENTED** [review]
Just left two minor comments.

File: lib/charms/operator_libs_linux/v1/snap.py [hunk 2]
```diff
@@ -354,6 +364,32 @@ def logs(self, services: Optional[List[str]] = None, num_lines: Optional[int] =
         args = ["logs", "-n={}".format(num_lines)] if num_lines else ["logs"]
         return self._snap_daemons(args, services).stdout
 
+    def connect(
+        self, plug: Optional[str] = None, service: Optional[str] = None, slot: Optional[str] = None
```
Comment (Mehdi-Bendriss) [inline lib/charms/operator_libs_linux/v1/snap.py:368]: It seems that the `plug` should maybe not be `Optional` here

File: tests/integration/helpers.py [hunk 3]
```diff
@@ -161,3 +165,61 @@ def check_tls(ip: str, port: int) -> bool:
...
+        shell=True,
+        universal_newlines=True,
+    ).splitlines()
+
+    logger.debug(f"{logs=}")
+
+    passed = False
+    for log in logs:
+        if topic and "index" in log:
+            passed = True
```
Comment (Mehdi-Bendriss) [inline tests/integration/helpers.py:223]: Maybe add a `break` after this?

**Review by Mehdi-Bendriss — COMMENTED** [review]
LGTM, just left some minor comments

**Pull request merged**